src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/VacancyTitle/VacancyApiTitlesProvider.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Services;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Recruit.Vacancies.Client.Infrastructure.Configuration;
+
+namespace Recruit.Vacancies.Client.Infrastructure.Services.VacancyTitle
+{
+    public class VacancyApiTitlesProvider : IGetVacancyTitlesProvider
+    {
+        private readonly string _vacancyApiReadSubscriptionKey;
+        private readonly ILogger<VacancyApiTitlesProvider> _logger;
+        private readonly string _baseSearchEndpoint;
+        private const string SubscriptionKeyHeaderKey = "Ocp-Apim-Subscription-Key";
+        private const int MaxSearchResultsPerPageLimit = 250;
+        private const int NoOfPagesToSearch = 4;
+
+        public VacancyApiTitlesProvider(ILogger<VacancyApiTitlesProvider> logger, IOptions<VacancyApiConfiguration> vacancyApiConfig)
+        {
+            _logger = logger;
+            _baseSearchEndpoint = vacancyApiConfig.Value.ApiSearchBaseUrl;
+            _vacancyApiReadSubscriptionKey = vacancyApiConfig.Value.ApiReadSubscriptionKey;
+        }
+
+        public async Task<IList<string>> GetVacancyTitlesAsync(string larsCode)
+        {
+            var frameworkCodeIdentifier = "-";
+            var isFrameworkCode = larsCode.Contains(frameworkCodeIdentifier);
+
+            if (isFrameworkCode)
+                larsCode = larsCode.Substring(0, larsCode.IndexOf(frameworkCodeIdentifier));
+
+            var vacancyApiSearchUrlFormat = $"{_baseSearchEndpoint}?{(isFrameworkCode ? "frameworkLarsCodes" : "standardLarsCodes")}={{0}}&pageSize={{1}}&pageNumber={{2}}";
+
+            var titles = new List<string>();
+
+            using (var httpClient = new HttpClient())
```

Comment:
Should we be using the ```IHttpClientFactory```? I'd assume we'd have the same issue as we've always had here with Dispose of HttpClient not explicitly freeing up connections.

Reply:
I tried to keep this simple but I can use the HttpClientFactory if you want.

Reply:
Using a static client for now.